app/ante/evm/04_validate_test.go

```diff
@@ -214,3 +220,70 @@ func getTxByType(typeTx string, recipient common.Address) evmtypes.EvmTxArgs {
 		panic("invalid type")
 	}
 }
+
+func (suite *EvmAnteTestSuite) TestCheckTxFee() {
+	// amount represents 1 token in the 18 decimals representation.
+	amount := math.NewInt(1_000_000_000_000_000_000)
+	gasLimit := uint64(1_000_000)
+
+	testCases := []struct {
+		name       string
+		txFee      *big.Int
+		txGasLimit uint64
+		expError   error
+	}{
+		{
+			name:       "pass",
+			txFee:      big.NewInt(amount.Int64()),
+			txGasLimit: gasLimit,
+			expError:   nil,
+		},
+		{
+			name:       "fail: not enough tx fees",
+			txFee:      big.NewInt(amount.Int64() - 1),
+			txGasLimit: gasLimit,
+			expError:   errortypes.ErrInvalidRequest,
+		},
+	}
+
+	for _, decimals := range []evmtypes.Decimals{
+		evmtypes.SixDecimals,
+		evmtypes.EighteenDecimals,
+	} {
+		for _, tc := range testCases {
+			suite.Run(fmt.Sprintf("%d decimals, %s", decimals, tc.name), func() {
+				// Call the configurator to set the EVM coin required for the
+				// function to be tested.
+				configurator := evmtypes.NewEVMConfigurator()
+				configurator.ResetTestConfig()
+				suite.Require().NoError(configurator.WithEVMCoinInfo(types.BaseDenom, uint8(decimals)).Configure())
+
+				// If decimals is not 18 decimals, we have to convert txFeeInfo to original
+				// decimals representation.
+				originalAmount := amount
+				if decimals != evmtypes.EighteenDecimals {
+					evmCoinDecimal := evmtypes.GetEVMCoinDecimals()
+					originalAmount = originalAmount.Quo(evmCoinDecimal.ConversionFactor())
+				}
```

Comment:
```suggestion
				originalAmount := amount
			        evmCoinDecimal := evmtypes.GetEVMCoinDecimals()
			        originalAmount = originalAmount.Quo(evmCoinDecimal.ConversionFactor())
```

Dont need to check, since conversion factor is modified accordingly